src/mesh/Router.cpp

```diff
@@ -316,15 +316,18 @@ bool perhapsDecode(meshtastic_MeshPacket *p)
         config.device.rebroadcast_mode == meshtastic_Config_DeviceConfig_RebroadcastMode_ALL_SKIP_DECODING)
         return false;
 
-    if (p->which_payload_variant == meshtastic_MeshPacket_decoded_tag)
-        return true; // If packet was already decoded just return
-
     // assert(p->which_payloadVariant == MeshPacket_encrypted_tag);
 
     // Try to find a channel that works with this hash
     for (ChannelIndex chIndex = 0; chIndex < channels.getNumChannels(); chIndex++) {
         // Try to use this hash/channel pair
         if (channels.decryptForHash(chIndex, p->channel)) {
+            if (p->which_payload_variant == meshtastic_MeshPacket_decoded_tag) {
+                // packet is not encoded, just set channel and return
+                p->channel = chIndex; 
+                return true;
+            }
+
```

Comment:
for decoded MeshPackets `p->channel` is already channel index, so `decryptForHash()` won't work since it can't match parameters (p->channel isn't channelHash):

[bool Channels::decryptForHash(ChannelIndex chIndex, ChannelHash channelHash)](https://github.com/meshtastic/firmware/blob/master/src/mesh/Channels.cpp#L294)